components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/builders/X509CredentialImpl.java

```diff
@@ -43,8 +61,153 @@
 public class X509CredentialImpl implements X509Credential {
 
     private PublicKey publicKey = null;
+    private PrivateKey privateKey = null;
     private X509Certificate signingCert = null;
 
+    private static KeyStore superTenantSignKeyStore = null;
+
+    private static Log log = LogFactory.getLog(X509CredentialImpl.class);
+
+    public static final String SECURITY_SAML_SIGN_KEY_STORE_LOCATION = "Security.SAMLSignKeyStore.Location";
+    public static final String SECURITY_SAML_SIGN_KEY_STORE_TYPE = "Security.SAMLSignKeyStore.Type";
+    public static final String SECURITY_SAML_SIGN_KEY_STORE_PASSWORD = "Security.SAMLSignKeyStore.Password";
+    public static final String SECURITY_SAML_SIGN_KEY_STORE_KEY_ALIAS = "Security.SAMLSignKeyStore.KeyAlias";
+    public static final String SECURITY_SAML_SIGN_KEY_STORE_KEY_PASSWORD = "Security.SAMLSignKeyStore.KeyPassword";
+
+    /**
+     * Instantiates X509Credential.
+     * This credential object will hold the private key, public key and the cert for the respective tenant domain.
+     *
+     * @param tenantDomain tenant domain
+     */
+    public X509CredentialImpl(String tenantDomain) throws IdentityException {
+
+        X509Certificate cert = null;
```

Comment:
This method seems too long. Can we break this to few methods in a readable way ?